### PR TITLE
[Flang][OpenMP] Use a label to avoid complexity of too many CHECK-DAGs in a test

### PR DESCRIPTION
**Pull request by kiranchandramohan**
This speeds up the test.

### CODE REVIEW

**Review by tblah — COMMENTED** [review]
How much does this speed up the test? It doesn't look to me like a particularly long test

+1 for this change if the difference is significant

**Comment by kiranchandramohan** [comment]
Speedup is 4.71s -> 0.27s on the machine I run.

I see that this test has failed in the CI. Will have a look.

**Review by tblah — APPROVED** [review]
looks good to me once this passes CI. Thanks!

**Comment by kiranchandramohan** [comment]
> > Speedup is 4.71s -> 0.27s on the machine I run.
> > I see that this test has failed in the CI. Will have a look.
> 
> Was this measured on a Windows box? If so, the new time looks small enough. If not, maybe the other CHECK-DAGs could be replaced as well, except for those that check global data definitions.

It was on a linux machine. I started with removing everything except the globals but the CI failed. So I added a few more for the threadprivates. May be there is something in the codegen of threadprivates that we can make more deterministic.

**Comment by luporl** [comment]
> > > Speedup is 4.71s -> 0.27s on the machine I run.
> > > I see that this test has failed in the CI. Will have a look.
> > 
> > 
> > Was this measured on a Windows box? If so, the new time looks small enough. If not, maybe the other CHECK-DAGs could be replaced as well, except for those that check global data definitions.
> 
> It was on a linux machine. I started with removing everything except the globals but the CI failed. So I added a few more for the threadprivates. May be there is something in the codegen of threadprivates that we can make more deterministic.

Oh, ok, that's interesting.

**Review by luporl — APPROVED** [review]
LGTM.

File: flang/test/Lower/OpenMP/threadprivate-real-logical-complex-derivedtype.f90 [hunk 1]
```diff
@@ -22,26 +22,26 @@ module test
 
 contains
   subroutine sub()
```
Comment (harishch4) [inline flang/test/Lower/OpenMP/threadprivate-real-logical-complex-derivedtype.f90:24]: Will adding a "CHECK-LABEL" here improves time?

Comment (kiranchandramohan) [reply flang/test/Lower/OpenMP/threadprivate-real-logical-complex-derivedtype.f90:24]: Thanks. That seems to work fine. I will update the patch.

Comment (kiranchandramohan) [reply flang/test/Lower/OpenMP/threadprivate-real-logical-complex-derivedtype.f90:25]: updated.

**Review by harishch4 — APPROVED** [review]

**Pull request merged**